src/mongo/delegates/Updates.class.php

```diff
@@ -229,19 +235,26 @@ public function saveChanges(
     protected function setReadPreferenceToPrimary()
     {
         // Set db preference
-        $dbPref = $this->getDatabase()->getReadPreference();
-        if($dbPref['type'] !== \MongoClient::RP_PRIMARY){
-            $this->originalDbReadPreference = $this->db->getReadPreference();
-            $tagsets = (isset($dbPref['tagsets']) ? $dbPref['tagsets'] : array());
-            $this->db->setReadPreference(\MongoClient::RP_PRIMARY, $tagsets);
+        /** @var ReadPreference $dbReadPref */
+        $dbReadPref = $this->getDatabase()->__debugInfo()['readPreference'];
```

Comment:
Interesting that debug info is a magic method here? can we rely on this? sorry what I mean is, clearly its meant to be used only for debugging, which means it could change massively btw versions of the driver, but what seem to be reliant on it?